### PR TITLE
feat: add support for npm packages

### PR DESCRIPTION
**Pull request by PierreDemailly**
Also removed case sensitivity for a given GitHub repository.

Fixes https://github.com/NodeSecure/cli/issues/207

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -47,10 +53,40 @@ export async function result(
...
+
+  try {
+    // We try to retrieve the repository with the GitHub API because
+    // Scorecard API is case sensitive, i.e if we pass "nodesecure/cli",
+    // we must reformat it to "NodeSecure/cli"
+    const { data } = await octokit.rest.repos.get({
+      owner,
+      repo
+    });
+    formattedRepository = data.full_name;
```
Comment (fraxken) [inline src/index.ts:69]: Would be cool to remove Octokit if this is just for one endpoint (introduce a lot of dependency for almost no value).

File: src/index.ts [hunk 2]
```diff
@@ -47,10 +53,40 @@ export async function result(
...
+      const latestVersion = data["dist-tags"].latest;
+      if (latestVersion) {
+        const repository = data.versions[latestVersion].repository;
+        const url = typeof repository === "string" ? repository : repository?.url;
+        formattedRepository = repositoryFromUrl(url ?? "");
+      }
+    }
+    catch {
+      throw new Error("Invalid repository, cannot find it on GitHub or NPM registry");
+    }
```
Comment (fraxken) [inline src/index.ts:85]: Maybe create a seperated method for that?

File: src/utils/repositoryFromUrl.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// https://docs.npmjs.com/cli/v9/configuring-npm/package-json#repository
+export function repositoryFromUrl(url: string): string {
+  if (url.startsWith("gist:") || url.includes("gist.github.com")) {
```
[the remark this answers is not in the document]
Comment (fraxken) [reply src/utils/repositoryFromUrl.ts:3]: We should maybe deal with a WHATWG here no? Would be easier

[the remark this answers is not in the document]
Comment (PierreDemailly) [reply src/utils/repositoryFromUrl.ts:3]: Yeah, it was the original plan but the fact that url can just be `foo/bar` make it throws as it's an invalid URL.
I'll deal with `try-catch` i think

**Review by fraxken — APPROVED** [review]

**Pull request merged**